storage/storage-interface/src/cached_state_view.rs

```diff
@@ -133,34 +123,33 @@ impl CachedStateView {
         // Do most of the work outside the write lock.
         let key_hash = state_key.hash();
         let state_value_option = match self.speculative_state.get(key_hash) {
-            StateStoreStatus::ExistsInScratchPad(blob) => Some(blob),
+            StateStoreStatus::ExistsInScratchPad(value) => Some(value),
             StateStoreStatus::DoesNotExist => None,
             // No matter it is in db or unknown, we have to query from db since even the
             // former case, we don't have the blob data but only its hash.
             StateStoreStatus::ExistsInDB | StateStoreStatus::Unknown => {
-                let (state_value, proof) = match self.latest_persistent_version {
-                    Some(version) => self
-                        .proof_fetcher
-                        .fetch_state_value_and_proof(state_key, version)?,
-                    None => (None, Some(SparseMerkleProof::new(None, vec![]))),
-                };
-                if let Some(proof) = proof {
-                    proof
-                        .verify(
-                            self.latest_persistent_state_root,
-                            key_hash,
-                            state_value.as_ref(),
-                        )
-                        .map_err(|err| {
-                            format_err!(
-                                "Proof is invalid for key {:?} with state root hash {:?}: {}",
-                                state_key,
-                                self.latest_persistent_state_root,
-                                err
-                            )
-                        })?;
+                match self.persisted_checkpoint {
+                    Some((version, root_hash)) => {
+                        let (value, proof) = self
+                            .proof_fetcher
+                            .fetch_state_value_and_proof(state_key, version)?;
+                        // TODO: proof verification can be opted out, for performance
```

Comment:
Do you see any downside of this? I actually want to get rid of this as a part of the async proof fetching change.

Reply:
Yeah I think we can make it optional, and when enabled, just log/panic on failure (for debugging purposes)